src/services/services.ts

```diff
@@ -1665,6 +1665,11 @@ namespace ts {
             return true;
         }
 
+        function isValidSnippetInsertionAtPosition(fileName: string, position: number): boolean {
+            const completionInfo = getCompletionsAtPosition(fileName, position);

```

Comment:
current implementation is effectively computes set of completions for the second time which is a bit too heavyweight given that it should be possible to provide answer for the question `isValidSnippetInsertionAtPosition` using only syntactic information and don't touch semantic data at all.